timeseer/podman/roles/traefik/tasks/main.yml

```diff
@@ -23,41 +23,50 @@
     mode: "0755"
     state: directory
 
+- name: Set ports configuration
+  ansible.builtin.set_fact:
+    ports_list:
+      - "80:80"
+      - "443:443"
+      - "8443:8443"
+
+- name: Define traefik_tls
+  ansible.builtin.set_fact:
+    acme_or_certs: "{{ 'acme' if traefik_tls == 'acme' else 'certs' }}"
+
+- name: Define traefik volume list
+  ansible.builtin.set_fact:
+    traefik_volume_list:
+      - "{{ traefik_dir }}/traefik.yml:/etc/traefik/traefik.yml"
+      - "{{ traefik_dir }}/traefik-timeseer.yml:/etc/traefik/traefik-timeseer.yml"
+      - "{{ traefik_dir }}/{{ acme_or_certs }}:/etc/traefik/{{ acme_or_certs }}{{ ':z' if ansible_selinux.status == 'enabled' else '' }}"
+
+- name: Prepend 'systemd-' to network
+  ansible.builtin.set_fact:
+    traefik_network: "systemd-{{ traefik_network }}"
+
 - name: Template configuration files
   ansible.builtin.template:
-    src: "{{ item }}"
-    dest: "{{ traefik_dir }}/{{ item }}"
+    src: "{{ item.src }}"
+    dest: "{{ item.dest }}"
     mode: "0644"
   loop:
-    - "traefik.yml"
-    - "traefik-timeseer.yml"
+    - src: "traefik.yml"
+      dest: "{{ traefik_dir }}/traefik.yml"
+    - src: "traefik-timeseer.yml"
+      dest: "{{ traefik_dir }}/traefik-timeseer.yml"
+    - src: "traefik-container.quadlet"
+      dest: "/etc/containers/systemd/traefik.container"
   register: traefik_config
-  notify:
-    - Signal Traefik Restart
-    - Restart Traefik
 
-- name: Run traefik  in a podman container
-  containers.podman.podman_container:
+- name: Reload systemd daemon to ensure traefik container is known
+  ansible.builtin.systemd:
+    daemon_reload: true
+  when: traefik_config.changed # noqa: no-handler
```

Comment:
There's no need to reload systemd when the Traefik configuration changes. It's best to split up the templating task in two tasks: one for the configuration, another for the container.